src/key_value_store/client/kvstore_client.cpp

```diff
@@ -0,0 +1,68 @@
+#include "kvstore_client.h"
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+bool KeyValueStoreClient::Put(const std::string &key,
+                              const std::string &value) {
+  PutRequest request;
+  PutReply response;
+  ClientContext context;
+  request.set_key(key);
+  request.set_value(value);
+
+  // TODO: ASK WHY THE REQ OBJ IS SENT INSTEAD OF PTR
```

Comment:
Also @barath  @adamegyed  I was really confused about this. Even though the type in the function is a PutRequest*, we are passing in a request object. Intuitively I'd say this is wrong but this is how Google did it in their example so...